src/onelogin/saml2/authn_request.py

```diff
@@ -47,9 +47,15 @@ def __init__(self, settings, force_authn=False, is_passive=False):
 
         destination = idp_data['singleSignOnService']['url']
 
-        name_id_policy_format = sp_data['NameIDFormat']
-        if 'wantNameIdEncrypted' in security and security['wantNameIdEncrypted']:
-            name_id_policy_format = OneLogin_Saml2_Constants.NAMEID_ENCRYPTED
+        name_id_policy_format = None
+        if 'wantNameId' in security and security['wantNameId']:
+            name_id_policy_format = sp_data['NameIDFormat']
+            if 'wantNameIdEncrypted' in security and security['wantNameIdEncrypted']:
+                name_id_policy_format = OneLogin_Saml2_Constants.NAMEID_ENCRYPTED
+
+        name_id_policy = ''
+        if name_id_policy_format: 
+            name_id_policy = """<samlp:NameIDPolicy Format="%(name_id_policy)s" AllowCreate="true" />""" % name_id_policy_format
```

Comment:
Ok, this is something that I will consider in a future release, to be able to not define NameIDPolicy element.